docs/en/creating-widgets/supplemental.md

```diff
@@ -0,0 +1,869 @@
+					onFocus={() => this.onFocus(3)}
+				/>
+				<FocusInputChild
+					key={4}
+					focus={this.currentlyFocusedKey === 4 ? this.shouldFocus : undefined}
+					onFocus={() => this.onFocus(4)}
+				/>
+			</div>
+		);
+	}
+}
+```
+
+# Managing state
+
+For simple applications where data is not required to flow between many components, state management can be straightforward to deal with. Data can be encapsulated within individual widgets that need it as the most [basic form of state management](#basic-internal-widget-state) within a Dojo application.
+
+As applications grow in complexity and start requiring data to be shared and transferred between multiple widgets, a more robust form of state management is required. Here, Dojo begins to prove its value as a reactive framework, allowing applications to define how data should flow between components, then letting the framework manage change detection and re-rendering. This is done by [wiring widgets and properties together](#intermediate-widget-properties) when declaring VDOM output in a widget's render function.
+
+For large applications, state management can be one of the most challenging aspects to deal with, requiring developers to balance between data consistency, availability and fault tolerance. While a lot of this complexity remains outside the scope of the web application layer, Dojo provides further solutions that help ensure data consistency. The [Dojo Stores](../dojo-stores/supplemental.md) component provides a centralized state store with a consistent API for accessing and managing data from multiple locations within the application.
+
+## Basic: self-encapsulated widget state
+
+Widgets can maintain their own internal state as locally-scoped variables.
+
+Internal state data may directly affect the widget's render output, or may be passed as properties to any child widgets where they in turn directly affect the children's render output. Widgets may also allow their internal state to be changed, for example in response to a user interaction event.
+
+The following example illustrates these patterns:
+
+> src/widgets/MyEncapsulatedStateWidget.tsx
+
+**Function-based variant:**
+
+```tsx
+import { create, tsx } from '@dojo/framework/core/vdom';
+
+const factory = create();
+
+let myState: string = 'Hello from a stateful widget!';
+let counter: number = 0;
+
+export default factory(function MyEncapsulatedStateWidget() {
+	return (
+		<div>
+			Current widget state: {myState}
+			<br />
+			<button
+				onclick={() => {
+					myState = 'State change iteration #' + ++counter;
+				}}
+			>
+				Change State
+			</button>
+		</div>
+	);
+});
+```
+
+**Class-based variant:**
+
+```tsx
+import WidgetBase from '@dojo/framework/core/WidgetBase';
+import { tsx } from '@dojo/framework/core/vdom';
+
+export default class MyEncapsulatedStateWidget extends WidgetBase {
+	private myState: string = 'Hello from a stateful widget!';
+	private counter: number = 0;
```

Comment:
In later TypeScript you don't need to type variables that are being set to primitives like `string` and `number`